lib/index.js

```diff
@@ -28,78 +27,81 @@ internals.hawk = function (server, options) {
     Hoek.assert(options, 'Invalid hawk scheme options');
     Hoek.assert(options.getCredentialsFunc, 'Missing required getCredentialsFunc method in hawk scheme configuration');
 
-    var settings = Hoek.clone(options);
+    const settings = Hoek.clone(options);
     settings.hawk = settings.hawk || {};
 
-    var scheme = {
-        authenticate: function (request, reply) {
-
-            Hawk.server.authenticate(request.raw.req, settings.getCredentialsFunc, settings.hawk, function (err, credentials, artifacts) {
+    const scheme = {
+        authenticate: async function (request, h) {
 
-                if (!err) {
-                    request.once('peek', function (chunk) {
+            let credentials;
+            let artifacts;
 
-                        var payloadHash = Hawk.crypto.initializePayloadHash(request.auth.credentials.algorithm, request.headers['content-type']);
-                        payloadHash.update(chunk);
-
-                        request.on('peek', function (chunk2) {
+            try {
+                ({ credentials, artifacts } = await Hawk.server.authenticate(request.raw.req, settings.getCredentialsFunc, settings.hawk));
```

Comment:
Use `var` inside try...catch to provide access to the variables.

Reply:
I forgot all about `var`